Failed to init app astro:audit, error: TypeError: Cannot read properties of undefined (reading 'trim')

### Astro Info

```block
Astro                    v4.3.6
Node                     v20.9.0
System                   macOS (arm64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             @astrojs/mdx
                         @astrojs/sitemap
                         @sentry/astro
                         @spotlightjs/astro
                         @astrojs/tailwind
```


### If this issue only occurs in one browser, which browser is a problem?

Chrome

### Describe the Bug

Using SVG link:
```html
<a aria-label="linkedin" href="/blog/linkedin" target="_blank"><text dy=".35em" style="fill: rgb(202, 106, 12); text-anchor: end;" transform="translate(-135,107.58290711686503)">linkedin</text></a>
```

https://developer.mozilla.org/en-US/docs/Web/SVG/Element/a

Error:

```sh
04:23:59 [ERROR] [toolbar] Failed to initialize dev toolbar app Audit (astro:audit):
TypeError: Cannot read properties of undefined (reading 'trim')
    at Object.match (node_modules/.pnpm/astro@4.3.6_@types+node@20.10.6_typescript@5.3.3/node_modules/astro/dist/runtime/client/dev-toolbar/apps/audit/a11y.js?v=da922a18:309:43)
    at lint (node_modules/.pnpm/astro@4.3.6_@types+node@20.10.6_typescript@5.3.3/node_modules/astro/dist/runtime/client/dev-toolbar/apps/audit/index.js?v=da922a18:68:22)
    at async Object.init (node_modules/.pnpm/astro@4.3.6_@types+node@20.10.6_typescript@5.3.3/node_modules/astro/dist/runtime/client/dev-toolbar/apps/audit/index.js?v=da922a18:28:5)
    at async AstroDevToolbar.initApp (node_modules/.pnpm/astro@4.3.6_@types+node@20.10.6_typescript@5.3.3/node_modules/astro/dist/runtime/client/dev-toolbar/toolbar.js?v=da922a18:323:7)
```

In:
```ts
{
    code: "a11y-missing-content",
    title: "Missing content",
    message: "Headings and anchors must have an accessible name, which can come from: inner text, aria-label, aria-labelledby, an img with alt property, or an svg with a tag <title></title>.",
    selector: a11y_required_content.join(","),
    match(element) {
      const innerText = element.innerText.trim(); // <--------------------
      if (innerText !== "")
```

Why?

- innerText is not defined for `svg > a`

Full SVG:
```html
<svg width="100%" viewBox="0 0 900 270" preserveAspectRatio="xMidYMin slice" id="the-chart"><g class="outerG" transform="translate(450,135)"><g class="slices"><path style="fill: rgb(78, 121, 167);" class="slice" d="M0,-108A108,108,0,0,1,50.19,-95.629L25.095,-47.815A54,54,0,0,0,0,-54Z"></path><path style="fill: rgb(242, 142, 44);" class="slice" d="M50.19,-95.629A108,108,0,0,1,25.846,104.862L12.923,52.431A54,54,0,0,0,25.095,-47.815Z"></path><path style="fill: rgb(202, 106, 12);" class="slice" d="M25.846,104.862A108,108,0,0,1,-100.982,38.297L-50.491,19.149A54,54,0,0,0,12.923,52.431Z"></path><path style="fill: rgb(199, 35, 38);" class="slice" d="M-100.982,38.297L-50.491,19.149Z"></path><path style="fill: rgb(225, 87, 89);" class="slice" d="M-100.982,38.297A108,108,0,0,1,-107.213,-13.018L-53.606,-6.509A54,54,0,0,0,-50.491,19.149Z"></path><path style="fill: rgb(118, 183, 178);" class="slice" d="M-107.213,-13.018A108,108,0,0,1,0,-108L0,-54A54,54,0,0,0,-53.606,-6.509Z"></path></g><g class="labels"><a aria-label="this site" href="/blog/this-site" target="_blank"><text dy=".35em" style="fill: rgb(78, 121, 167); text-anchor: start;" transform="translate(135,-117.96943081726532)">this site</text></a><a aria-label="youtube channel" href="/blog/youtube" target="_blank"><text dy=".35em" style="fill: rgb(242, 142, 44); text-anchor: start;" transform="translate(135,14.645206651021738)">youtube channel</text></a><a aria-label="linkedin" href="/blog/linkedin" target="_blank"><text dy=".35em" style="fill: rgb(202, 106, 12); text-anchor: end;" transform="translate(-135,107.58290711686503)">linkedin</text></a><a aria-label="conference" href="/blog/conference" target="_blank"><text dy=".35em" style="fill: rgb(199, 35, 38); text-anchor: end;" transform="translate(-135,43.08449377566811)">conference</text></a><a aria-label="interview" href="/blog/interview" target="_blank"><text dy=".35em" style="fill: rgb(225, 87, 89); text-anchor: end;" transform="translate(-135,14.645206651021805)">interview</text></a><a aria-label="only external projects" href="#" target="_blank"><text dy=".35em" style="fill: rgb(118, 183, 178); text-anchor: end;" transform="translate(-135,-90.94405590278875)">only external projects</text></a></g><g class="lines"><polyline points="19.384568807292183,-78.64628721151021,29.076853210938275,-117.96943081726532,128.25,-117.96943081726532"></polyline><polyline points="80.40941880194237,9.763471100681159,120.61412820291356,14.645206651021738,128.25,14.645206651021738"></polyline><polyline points="-37.642576935545215,71.72193807791002,-56.463865403317826,107.58290711686503,-128.25,107.58290711686503"></polyline><polyline points="-75.73631565751859,28.722995850445404,-113.60447348627788,43.08449377566811,-128.25,43.08449377566811"></polyline><polyline points="-80.40941880194237,9.763471100681205,-120.61412820291356,14.645206651021805,-128.25,14.645206651021805"></polyline><polyline points="-53.71293531750444,-60.62937060185917,-80.56940297625665,-90.94405590278875,-128.25,-90.94405590278875"></polyline></g></g><g class="innerG" transform="translate(450,135)"><g class="slices"><path style="fill: #4e79a7;" class="slice" d="M0,-40.5A40.5,40.5,0,0,1,18.821,-35.861L6.274,-11.954A13.5,13.5,0,0,0,0,-13.5Z"></path><path style="fill: #f28e2c;" class="slice" d="M18.821,-35.861A40.5,40.5,0,1,1,-37.868,14.361L-12.623,4.787A13.5,13.5,0,1,0,6.274,-11.954Z"></path><path style="fill: #e15759;" class="slice" d="M-37.868,14.361A40.5,40.5,0,0,1,-40.205,-4.882L-13.402,-1.627A13.5,13.5,0,0,0,-12.623,4.787Z"></path><path style="fill: #76b7b2;" class="slice" d="M-40.205,-4.882A40.5,40.5,0,0,1,0,-40.5L0,-13.5A13.5,13.5,0,0,0,-13.402,-1.627Z"></path></g><g class="labels"><a aria-label="blogging" href="/blog/blogging" target="_blank"><text dy=".35em" style="fill: #4e79a7; text-anchor: start;" transform="translate(135,-39.32314360575511)">blogging</text></a><a aria-label="social media" href="/blog/social-media" target="_blank"><text dy=".35em" style="fill: #f28e2c; text-anchor: start;" transform="translate(135,30.314685300929586)">social media</text></a><a aria-label="public speaking" href="/blog/public-speaking" target="_blank"><text dy=".35em" style="fill: #e15759; text-anchor: end;" transform="translate(-135,4.881735550340602)">public speaking</text></a><a aria-label="open-source contributions" href="/blog/open-source-contributions" target="_blank"><text dy=".35em" style="fill: #76b7b2; text-anchor: end;" transform="translate(-135,-30.314685300929586)">open-source contributions</text></a></g><g class="lines"><polyline points="6.461522935764061,-26.215429070503404,9.692284403646092,-39.32314360575511,128.25,-39.32314360575511"></polyline><polyline points="17.904311772501476,20.20979020061972,26.856467658752216,30.314685300929586,128.25,30.314685300929586"></polyline><polyline points="-26.80313960064746,3.2544903668937346,-40.20470940097118,4.881735550340602,-128.25,4.881735550340602"></polyline><polyline points="-17.90431177250148,-20.20979020061972,-26.85646765875222,-30.314685300929586,-128.25,-30.314685300929586"></polyline></g></g></svg>
```

### What's the expected result?

no error

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-5z8nnr?file=src%2Fpages%2Findex.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

--- a/.changeset/spotty-cobras-warn.md
+++ b/.changeset/spotty-cobras-warn.md
@@ -1,0 +1,5 @@
+---
+"astro": patch
+---
+
+Fixes an issue where audit fails to initialize when encountered `<a>` inside `<svg>`

--- a/packages/astro/src/runtime/client/dev-toolbar/apps/audit/a11y.ts
+++ b/packages/astro/src/runtime/client/dev-toolbar/apps/audit/a11y.ts
@@ -338,8 +338,8 @@ export const a11y: AuditRuleWithSelector[] = [
 		selector: a11y_required_content.join(','),
 		match(element: HTMLElement) {
 			// innerText is used to ignore hidden text
-			const innerText = element.innerText.trim();
-			if (innerText !== '') return false;
+			const innerText = element.innerText?.trim();
+			if (innerText && innerText !== '') return false;
 
 			// Check for aria-label
 			const ariaLabel = element.getAttribute('aria-label')?.trim();

